src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java

```diff
@@ -97,12 +100,72 @@ public void deleteAll() {
             return null;
         }
 
+        Assert.isTrue(results.size() == 1, "should be only 1 one vertex with given id");
+
         source.setGremlinResultReader(new GremlinResultVertexReader());
         source.setGremlinSourceReader(new GremlinSourceVertexReader());
-
         source.doGremlinResultRead(results.get(0));
-        Assert.isTrue(results.size() == 1, "should be only 1 one vertex with given id");
 
         return this.mappingConverter.read(domainClass, source);
     }
+
+    private <T> void completeEdge(@NonNull T domain, @NonNull GremlinSourceEdge source) {
+        Assert.isInstanceOf(GremlinSourceEdge.class, source, "should be instance of GremlinSourceEdge");
```

Comment:
Is it possible that source is not an instance of GremlinSourceEdge?

Reply:
No need here, will remove it.